src/Symfony/Component/Messenger/Middleware/SendMessageMiddleware.php

```diff
@@ -61,10 +61,9 @@ public function handle(Envelope $envelope, StackInterface $stack): Envelope
             } else {
                 /** @var RedeliveryStamp|null $redeliveryStamp */
                 $redeliveryStamp = $envelope->last(RedeliveryStamp::class);
-
                 $senders = \iterator_to_array($this->sendersLocator->getSenders($envelope, $handle));
```

Comment:
insteaf of this `iterator_to_array` + the `if` just below, shouldn't the event be sent in the foreach, after the `if` there, only the 1st time? it would naturally resolve the issue fixed here also, isn't it?

Reply:
I had that before (I set a boolean variable before the loop and used it to only dispatch once) and Sam wanted me to do it this way :). So, I'm cool with whichever way